### PR TITLE
Fix bug in cluster/log-dump

### PR DESCRIPTION
**Pull request by shyamjvs**
We're breaking in case KUBECTL is set as "./cluster/kubectl.sh --match-server-version". Moreover we always are using cluster/kubectl.sh as the default and don't want to do match-server-version for the purpose of logexporter.

Also adding owners file so I'm not blocked for approves while making fixes in log-dump. Besides I'll be able to review fixes sent by others.

/cc @wojtek-t

### CODE REVIEW

File: cluster/log-dump/OWNERS [hunk 1]
```diff
@@ -0,0 +1,5 @@
+reviewers:
+  - shyamjvs
+  - wojtek-t
```
Comment (wojtek-t) [inline cluster/log-dump/OWNERS]: I don't want to be reviewer here.

File: cluster/log-dump/log-dump.sh [hunk 2]
```diff
@@ -282,7 +282,7 @@ function dump_nodes_with_logexporter() {
   sed -i'' -e "s@{{.EnableHollowNodeLogs}}@${enable_hollow_node_logs}@g" "${KUBE_ROOT}/cluster/log-dump/logexporter-daemonset.yaml"
 
   # Create the logexporter namespace, service-account secret and the logexporter daemonset within that namespace.
-  KUBECTL="${KUBECTL:-${KUBE_ROOT}/cluster/kubectl.sh}"
+  KUBECTL="${KUBE_ROOT}/cluster/kubectl.sh"
```
Comment (wojtek-t) [inline cluster/log-dump/log-dump.sh]: Why this change? If KUBECTL is defined, we should use it.

File: cluster/log-dump/OWNERS (hunk 1 above)
Comment (shyamjvs) [inline cluster/log-dump/OWNERS]: Removed.

File: cluster/log-dump/log-dump.sh (hunk 2 above)
Comment (shyamjvs) [inline cluster/log-dump/log-dump.sh]: Sometimes KUBECTL is defined as `./cluster/kubectl.sh --match-server-version`. We use that flag for some tests and we wouldn't be able to run logexporter there due to version skew (even though we're able to create daemonset). Wdyt?

**Comment by shyamjvs** [comment]
/retest
(seems like a git fetch error)

**Comment by shyamjvs** [comment]
@wojtek-t Responded to comments. PTAL

File: cluster/log-dump/log-dump.sh (hunk 2 above)
Comment (wojtek-t) [inline cluster/log-dump/log-dump.sh]: Sorry, I don't understand. Yes, I know what KUBECT can be set to. But how is that problematic?

Comment (shyamjvs) [inline cluster/log-dump/log-dump.sh]: When it is set to "./cluster/kubectl.sh --match-server-version" (like here - https://github.com/kubernetes/test-infra/blob/master/kubetest/e2e.go#L33), logexporter would fail if there actually is a skew. IIUC we don't want (for the purpose of logexporter) to fail because of the skew, as we just care about uploading the logs (even if there's a skew).

**Comment by roberthbailey** [comment]
Rather than adding you an an owner to the log-dump directory within cluster (which we are actively trying to delete) can we come up with a plan to migrate it out of the main repo? It looks to me like it's a utility used primarily for e2e testing, so I'd suggest moving it to the test-infra repository. 

The script pulls in `kube-util.sh`; how much functionality from the bash scripts do you actually rely on?

**Comment by shyamjvs** [comment]
That sounds reasonable. However one strong reason to have it inside the main repo is because it's being used by our e2e testing framework which is independent of test-infra and is crucial to allow our e2e-testing be self-container within the repo. Otherwise we'll have to do something like checkout the test-infra repo or wget those scripts over the internet from within our e2e framework (which doesn't seem to be a very healthy dependency). I'd love to hear any clean solution for this!

**Comment by shyamjvs** [comment]
Moreover, we seem to have this policy of not vendoring anything from test-infra inside our main repo.

**Comment by wojtek-t** [comment]
@roberthbailey - I agree with @shyamjvs - this isn't really part of test-infra repo, bu more part of e2e framework. That said, we should try to move this script under test/e2e (it should be used only there).

@shyamjvs - can you please add a TODO to move it to test/e2e?

**Comment by shyamjvs** [comment]
@wojtek-t Done.

**Comment by shyamjvs** [comment]
Removed the owners file.
